agent/consul/state/catalog_test.go

```diff
@@ -3600,6 +3600,12 @@ func TestStateStore_CheckConnectServiceNodes_Gateways(t *testing.T) {
 	}, nil))
 	assert.True(watchFired(ws))
 
+	ws = memdb.NewWatchSet()
```

Comment:
This new addition to the test is so that we will make it into the for loop to run the `watchChans = append(watchChans, gwServices.WatchCh())`?

Reply:
No, the `CheckServiceNodes` function still isn't using those channels.

This is the bit it relies on now:
```go
// Ensure that blocking queries wake up if the gateway-service mapping exists, but the gateway does not exist yet
if !exists {
	ws.Add(gwServices.WatchCh())
}
```

We only add these channels when the gateway doesn't exist. For gateways that do exist we continue to use the channel from `maxIndexAndWatchChForService`.

Reply:
Now that I think about it again, other functions that add all the the watchChans to the WatchSet will now have some duplicates.

Will fix that.

Reply:
Do you think it is possible to get rid of the `watchChans` return value here entirely and some a similar watch pattern in the other places we use this function?

Adds a bit more complexity to call sites, but reduces return values from 4 to 3, and which is an improvement.

Reply:
Yeah, I'll take a pass at it and see how it comes out.